src/Controller/Images.php

```diff
@@ -89,6 +90,50 @@ private function getFilesIndex(string $path, string $type): Collection
         return new Collection($files);
     }
 
+    /**
+     * @Route("/article_files", name="bolt_article_files", methods={"GET"})
+     */
+    public function getFilesList(Request $request): JsonResponse
+    {
+        try {
+            $this->validateCsrf('bolt_article');
+        } catch (InvalidCsrfTokenException $e) {
+            return new JsonResponse([
+                'error' => true,
+                'message' => 'Invalid CSRF token',
+            ], Response::HTTP_FORBIDDEN);
+        }
+
+        $locationName = $this->request->query->get('location', 'files');
+        $type = $this->request->query->get('type', '');
+
+        $path = $this->config->getPath($locationName, true);
+
+        $files = $this->getFilesIndex($path, $type);
+
+        return new JsonResponse($files);
+    }
+
+    private function getFilesIndex(string $path, string $type): Collection
+    {
+        $fileTypes = $this->config->getFileTypes()->toArray();
+        $glob = '*.{' . implode(',', $fileTypes) . '}';
+
+        $files = [];
+
+        $textExtenion = new TextExtension();
+
+        foreach ($this->findFiles($path, $glob) as $file) {
+            $files[] = [
+                'title' => $file->getRelativePathname(),
+                'url' => '/files/' . $file->getRelativePathname(),
```

Comment:
Should we use the configured location for `files` here: https://github.com/bolt/core/blob/4.1/src/Configuration/FileLocations.php

instead of hardcoding `/files/`?

Reply:
yes, will do!

Reply:
On second thought.. The `$config->getPath('files')` gives the absolute path. We're using this in several places. We should implement a `$config->getUri('files')` method, that gives us `/files/`. Until that time, we can't really make it better than this.

Reply:
Right! Hmm, looks like the 2nd parameter of `getPath` can be set to `false`:
https://github.com/bolt/core/blob/a418e99fe7e66eaeb4b11fa386dd115c137e34eb/src/Configuration/Config.php#L203-L206 maybe?

Or maybe something like this: https://github.com/bolt/core/blob/113254f6f25678af2d0219e8e60b4b183a6d7ab0/src/DataFixtures/ContentFixtures.php#L68

I don't want to get too pushy though 😇

Reply:
No worries about being too pushy.. Both of those don't work, though.. They give: 

```
Images.php on line 131:
"/Users/bob/Sites/Bolt/playground/public/files"
```

Simply calling `basename()` on them might work, but feels icky.. We will need to do work on this eventually anyways, when we get around to implementing FlySystem support.

Reply:
👍 Yes, let's do that then!

Ready to merge, in that case?